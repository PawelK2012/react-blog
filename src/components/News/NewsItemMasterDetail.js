import React, { Component } from 'react'
import { Card, CardHeader, CardMedia, CardTitle, CardText } from 'material-ui/Card'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

class NewsItem extends Component {

    render() {
        return (
            <MuiThemeProvider>
                <Card className="card">
                    <CardHeader
                        title={this.props.title}
                        subtitle={this.props.subTitle}
                    />
                    <CardMedia
                        overlay={<CardTitle title={this.props.title} subtitle={this.props.subTitle} />}
                    >
                        <img src={this.props.img} alt={this.props.title} />
                    </CardMedia>
                    <CardTitle title={this.props.title} subtitle={this.props.subTitle} />
                    <CardText>{this.props.text}</CardText>
                </Card>
            </MuiThemeProvider>
        )
    }
}
export default NewsItem

