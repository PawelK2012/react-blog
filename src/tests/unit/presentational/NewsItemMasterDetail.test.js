import React from 'react';
import { shallow } from 'enzyme';
import NewsItemMasterDetail from '../../../presentational/News/NewsItemMasterDetail';

describe('NewsItemMasterDetail Component:', () => { 
    test('renders correctly', () => {
      const wrapper = shallow(<NewsItemMasterDetail />);
      expect(wrapper.html()).toMatchSnapshot();
    });
});