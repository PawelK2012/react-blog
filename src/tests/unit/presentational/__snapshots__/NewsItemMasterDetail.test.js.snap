// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItemMasterDetail Component: renders correctly 1`] = `
"<div class=\\"card\\" style=\\"color:rgba(0, 0, 0, 0.87);background-color:#ffffff;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12);border-radius:2px;z-index:1;mui-prepared:\\"><div style=\\"padding-bottom:8px\\"><div style=\\"padding:16px;font-weight:500;box-sizing:border-box;position:relative;white-space:nowrap;mui-prepared:\\"><div style=\\"display:inline-block;vertical-align:top;white-space:normal;padding-right:90px;mui-prepared:\\"><span style=\\"color:rgba(0, 0, 0, 0.87);display:block;font-size:15px;mui-prepared:\\"></span><span style=\\"color:rgba(0, 0, 0, 0.54);display:block;font-size:14px;mui-prepared:\\"></span></div></div><div style=\\"position:relative;mui-prepared:\\"><div style=\\"mui-prepared:\\"><img style=\\"vertical-align:top;max-width:100%;min-width:100%;width:100%;mui-prepared:\\"/></div><div style=\\"position:absolute;top:0;bottom:0;right:0;left:0;mui-prepared:\\"><div style=\\"height:100%;position:relative;mui-prepared:\\"><div style=\\"position:absolute;bottom:0;right:0;left:0;padding-top:8px;background:rgba(0, 0, 0, 0.54);mui-prepared:\\"><div style=\\"padding:16px;position:relative;mui-prepared:\\"><span style=\\"font-size:24px;color:rgba(255, 255, 255, 0.87);display:block;line-height:36px;mui-prepared:\\"></span><span style=\\"font-size:14px;color:rgba(255, 255, 255, 0.54);display:block;mui-prepared:\\"></span></div></div></div></div></div><div style=\\"padding:16px;position:relative;mui-prepared:\\"><span style=\\"font-size:24px;color:rgba(0, 0, 0, 0.87);display:block;line-height:36px;mui-prepared:\\"></span><span style=\\"font-size:14px;color:rgba(0, 0, 0, 0.54);display:block;mui-prepared:\\"></span></div><div style=\\"padding:16px;font-size:14px;color:rgba(0, 0, 0, 0.87);mui-prepared:\\"></div></div></div>"
`;
